Mathlib/LinearAlgebra/Dual.lean

```diff
@@ -1433,15 +1387,9 @@ theorem dualPairing_nondegenerate : (dualPairing K V₁).Nondegenerate :=
 #align linear_map.dual_pairing_nondegenerate LinearMap.dualPairing_nondegenerate
 
 theorem dualMap_surjective_of_injective {f : V₁ →ₗ[K] V₂} (hf : Function.Injective f) :
-    Function.Surjective f.dualMap := by
-  intro φ
-  let f' := LinearEquiv.ofInjective f hf
-  use Subspace.dualLift (range f) (f'.symm.dualMap φ)
-  ext x
-  rw [LinearMap.dualMap_apply, Subspace.dualLift_of_mem (mem_range_self f x),
-    LinearEquiv.dualMap_apply]
-  congr 1
-  exact LinearEquiv.symm_apply_apply f' x
+    Function.Surjective f.dualMap := fun φ ↦
+  have ⟨f', hf'⟩ := f.exists_leftInverse_of_injective (ker_eq_bot.mpr hf)
+  ⟨φ.comp f', ext fun x ↦ congr_arg φ (FunLike.congr_fun hf' x)⟩
```

Comment:
Something like this probably works. You might even be able to put a forall into the `congr(...)` and remove the `ext`.
```suggestion
  ⟨φ.comp f', ext fun x ↦ congr(φ $hf' x)⟩
```
This is not something you need to change for this PR though.

Reply:
Thanks! `congr(φ <| $hf' x)` works.